### PR TITLE
Encode SubjectAlternativeName extension

### PR DESCRIPTION
**Pull request by sree-revoori1**
Currently, only the otherName choice is supported.

### CODE REVIEW

File: platform/src/lib.rs [hunk 1]
```diff
@@ -125,4 +134,11 @@ pub trait Platform {
     ///
     /// Example: 99991231235959Z is December 31st, 9999 23:59:59 UTC
     fn get_cert_validity(&mut self) -> Result<CertValidity, PlatformError>;
+
+    /// Retrieves the SubjectAlternativeName extension
+    ///
+    /// Currently, only the otherName choice is supported. This function
+    /// can be left unimplemented if the SubjectAlternativeName extension is
+    /// not needed in the DPE leaf cert or CSR.
+    fn get_subject_alternative_name(&mut self) -> Result<OtherName, PlatformError>;
```
Comment (jhand2) [inline platform/src/lib.rs:143]: nit: Should this return a generic SubjectAltName enum where once of the options is OtherName? Then if we want to support the other types we don't need to change this API.

Comment (sree-revoori1) [reply platform/src/lib.rs]: Done

File: platform/src/default.rs [hunk 2]
```diff
@@ -197,4 +197,8 @@ impl Platform for DefaultPlatform {
             not_after: not_after_vec,
         })
     }
+
+    fn get_subject_alternative_name(&mut self) -> Result<SubjectAltName, PlatformError> {
```
Comment (jhand2) [inline platform/src/default.rs:201]: Might be good to populate this with something so we have test coverage/cert linting against the go code.

Comment (sree-revoori1) [reply platform/src/default.rs:201]: Done except I just added it to tests in x509.rs so it is easier to test without using certifyKey.

**Review by jhand2 — APPROVED** [review]

**Pull request merged**